File: analyze_template.py
import sys
from google.oauth2 import service_account
from googleapiclient.discovery import build

def main():
    creds_path = '/Users/yms/.gemini/Fabric_Master/bagorderapp-fe5c29e3e221.json'
    sheet_id = '116dTfVGk8tErpm-eCgjm3xdG3UCOjQeo'
    
    scopes = ['https://www.googleapis.com/auth/spreadsheets.readonly']
    
    try:
        creds = service_account.Credentials.from_service_account_file(creds_path, scopes=scopes)
        service = build('sheets', 'v4', credentials=creds)
        
        # Get spreadsheet metadata
        sheet_metadata = service.spreadsheets().get(spreadsheetId=sheet_id).execute()
        sheets = sheet_metadata.get('sheets', '')
        
        print(f"Spreadsheet Title: {sheet_metadata.get('properties', {}).get('title')}\n")
        
        for sheet in sheets:
            title = sheet.get("properties", {}).get("title", "")
            print(f"=== Sheet: {title} ===")
            
            # Get first 30 rows to understand template layout
            result = service.spreadsheets().values().get(
                spreadsheetId=sheet_id, range=f"'{title}'!A1:Z30").execute()
            
            rows = result.get('values', [])
            if not rows:
                print("No data found.")
                continue
                
            for idx, row in enumerate(rows):
                print(f"Row {idx+1}: {row}")
            print("\n")
            
    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)

if __name__ == '__main__':
    main()

File: backend/analyze_template.js
const { google } = require('googleapis');
const path = require('path');

const KEY_FILE_PATH = path.join(__dirname, '../bagorderapp-fe5c29e3e221.json');
const SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly'];
const auth = new google.auth.GoogleAuth({ keyFile: KEY_FILE_PATH, scopes: SCOPES });
const sheet_id = '116dTfVGk8tErpm-eCgjm3xdG3UCOjQeo';

async function main() {
  try {
    const client = await auth.getClient();
    const sheets = google.sheets({ version: 'v4', auth: client });
    
    const meta = await sheets.spreadsheets.get({ spreadsheetId: sheet_id });
    console.log(`Spreadsheet Title: ${meta.data.properties.title}\n`);
    
    for (const sheet of meta.data.sheets) {
      const title = sheet.properties.title;
      console.log(`=== Sheet: ${title} ===`);
      const res = await sheets.spreadsheets.values.get({
        spreadsheetId: sheet_id,
        range: `'${title}'!A1:Z30`
      });
      const rows = res.data.values;
      if (!rows) {
        console.log("No data found.");
        continue;
      }
      rows.forEach((row, idx) => {
        console.log(`Row ${idx+1}:`, JSON.stringify(row));
      });
      console.log("\n");
    }
  } catch(e) {
    console.error(e);
  }
}
main();
